python/GafferSceneTest/InstancerTest.py

```diff
@@ -202,7 +202,7 @@ def testEmptyName( self ) :
 		instancer["parent"].setValue( "/plane" )
 		instancer["name"].setValue( "" )
 
-		self.assertScenesEqual( instancer["out"], plane["out"] )
+		self.assertScenesEqual( instancer["out"], plane["out"], pathsToIgnore = ( "/plane", ) )
```

Comment:
This is necessary because we're deleting the target object without considering the name plug. I wasn't sure which way is desirable to be honest, but this approach keeps the affects/hash/compute simpler...

Reply:
I can see arguments either way, but unless someone wants to champion the "empty name doesn't delete the object" approach, I'm happy enough with this.

Reply:
I guess the danger might be that someone has deleted the name in order to disable an instancer, and we've broken their scene. Perhaps we should tie this in with the `deleteObject` plug discussion, so that if we do add that plug for compatibility, `processesRootObject()` becomes `deleteObjectPlug()->getValue() && !namePlug()->getValue().empty()`?